Tests/SwiftFormatTests/Rules/OrderedImportsTests.swift

```diff
@@ -651,4 +651,36 @@ final class OrderedImportsTests: LintOrFormatRuleTestCase {
       ]
     )
   }
+
+  func testImportsOrderWithFileIgnoreDirective() {
+    assertFormatting(
```

Comment:
Does this also end up adding a newline after `// swift-format-ignore-file` if no re-ordering is necessary? Eg. for

```swift
// swift-format-ignore-file
import LibA
import LibB
```

To be clear, I’m fine with that, just want to check and record the behavior in a test case.

--- 

One thing I did notice is that I think this flips comments order around if you have eg.

```swift
// We need to ignore this file because it is generated
// swift-format-ignore-line

import Foundation
```

Maybe `swift-format-ignore-line` just needs to flush the `commentBuffer` into the `headerBuffer` similar to what a blank line does.

Reply:
> Does this also end up adding a newline after // swift-format-ignore-file if no re-ordering is necessary?

Yes, that's right. In the current implementation,

```swift
// swift-format-ignore-file
import LibA
import LibB
```

is formatted as

```swift
// swift-format-ignore-file

import LibA
import LibB
```

because the `// swift-format-ignore-file` comment is treated as part of the file header and is separated accordingly.

Additionally, I have fixed the issue where the comment order could be flipped when`// swift-format-ignore-line` appeared after the first line, and I have added a test case for it.

Thanks for the feedback!